pkgs/development/tools/parsing/tree-sitter/grammar.nix

```diff
@@ -34,6 +35,12 @@ stdenv.mkDerivation ({
     tree-sitter generate
   '';
 
+  doCheck = !isBroken;
+  checkPhase = ''
+    # HOME is needed because tree-sitter needs a writable home folder
+    HOME=. ${lib.getExe tree-sitter} test
```

Comment:
```suggestion
    runHook preCheck

    # HOME is needed because tree-sitter needs a writable home folder
    HOME=. ${lib.getExe tree-sitter} test

    runHook postCheck
```